---
title: "Notropis_cytb_013024"
author: "K. Dye"
date: "2024-01-30"
output: html_document
---

#Obtain data
```{bash}
unzip raw_seq_kd.zip
```
## Should have a list of the ab1 files for the forward and reverse reads of each individual

#Geneious was used to align forward and reverse sequences (ab1 files) and create a consensus fasta file for each individual.
  #Some individuals were removed due to poor sequence quality
    # NM-116, NM-230, NM-259, and NM-320 were removed
  #Consensus sequences are in /data/Notorpis_consensus_seqs
  #Outgroup fasta sequences obtained from GenBank, found in /data/Notropis_outgroup_seqs

#Remove header from outgroup fasta files
```{bash}
sed -i '1s/^>.*/>Agosia chrysogaster/' agosia_chrysogaster.fasta
sed -i '1s/^>.*/>Notropis atherinoides/' notropis_atherinoides.fasta
sed -i '1s/^>.*/>Notropis hudsonius HUD1/' notropis_hud1.fasta
sed -i '1s/^>.*/>Notropis jemezanus JEME2/' notropis_jeme2.fasta
sed -i '1s/^>.*/>Notropis jemezanus/' notropis_jemezanus.fasta
sed -i '1s/^>.*/>Notropis atherinoides NAN1/' notropis_nan1.fasta
sed -i '1s/^>.*/>Notropis percobromus NPB1/' notropis_pb1.fasta
sed -i '1s/^>.*/>Notropis stilbius/' notropis_stilbius.fasta
sed -i '1s/^>.*/>Pimephales vigilax/' pimephales_vigilax.fasta
```

#Create multi fasta with all consensus and outgroup sequences

## Linearizing outgroup fasta files
```{bash}
for file in *.fasta; do
    sed -e 's/\(^>.*$\)/#\1#/' "$file" | tr -d "\r" | tr -d "\n" | sed -e 's/$/#/' | tr "#" "\n" | sed -e '/^$/d' > "${file%.fasta}_linear.fasta"
done
```

## Renaming of fasta files
```{bash}
cut -d " " -f1 Notropis_cytb_consensus.fasta | sed 's/Nucleotide_alignment_//g' > Notropis_cytb_consensus_rename.fasta
```

## Create multi fasta to be aligned
```{bash}
cat Notropis_cytb_consensus_rename.fasta ../outgroups/*linear.fasta > all.fasta
```

#Align sequences using ClustalOmega
```{bash}
../programs/bin/bin/clustalo -i all.fasta -o align.fasta --outfmt=fasta
```

## Check alignment manually using UGene
  ## Reimport alignment as ugene_alignment.fasta

#Trim
## Linearize multi-fasta for trimming
```{bash}
awk '/^>/ {if (seq != "") {print seq; seq="";} print; next} {seq = seq $0} END {if (seq != "") print seq}' ugene_alignment.fasta > linear_alignment.fasta
```

## Trim first 26 bases and last 29 bases off each sequence
```{bash}
awk '/^>/ {if (seq != "") {print substr(seq,27,length(seq)-55); seq="";} print; next} {seq = seq $0} END {if (seq != "") print substr(seq,27,length(seq)-55)}' linear_alignment.fasta > trimmed_alignment.fasta
```
*** Trimmed alignment contains 1,121 bp and 56 individuals + outgroups ***

## Renamed individuals manually for ease of identification
  ## Reimported as final_trim.fasta


#Run model test
```{bash}
modeltest竏地g 竏段 final_trim.fasta 
```      

#Phylogenetic Reconstruction - Bayesian analysis

#Beauti
 ## Run five times 
```{GUI}
Use Beauti to make xml
    Import alignment final_trim.fasta
    Use log file from modeltest
        
modeltest竏地g 竏段 final_trim.fasta 

Best model according to BIC
---------------------------
Model:              TrN+I+G4
lnL:                -4840.5020
Frequencies:        0.2600 0.3105 0.1406 0.2889
Subst. Rates:       1.0000 52.2687 1.0000 1.0000 9.1658 1.0000 
Inv. sites prop:    0.6185
Gamma shape:        3.4150
Score:              10621.9489
Weight:             0.5655

    Gamma category count : 6
    Shape : 3.4150 (estimate)
    Proportion invariant : 0.6185 (estimate)
    Subst Model : TN93 (everything estimated)
    
  MCMC 
    
    Store every 1000
    tracelog : notropis_1
    treelog : notropis_1
    
  File > save as > notropis_1.xml
``` 

#BEAST
## Run on each xml file made using Beauti
```{GUI}
open xml file : notropis_1
Run
```

#Tracer
```{GUI}
#look at log file from each BEAST run 
```

#Tree Annotator
```{GUI}
Maximum Clade Credibility Tree
Node heights : mean heights
```    

